Purchases/Purchasing/StoreKit2/ProductsFetcherSK2.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ProductsManagerSK2.swift
+//
+//  Created by Andrés Boedo on 7/23/21.
+
+import Foundation
+import StoreKit
+
+enum ProductsManagerSK2Error: Error {
+
+    case productsRequestError(innerError: Error)
+
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+actor ProductsFetcherSK2 {
+
+    private var cachedProductsByIdentifier: [String: SK2ProductDetails] = [:]
+
+    func products(identifiers: Set<String>) async throws -> Set<SK2ProductDetails> {
+        do {
+            // todo: also cache requests, so that if a request is in flight for the same products,
```

Comment:
Are these `todos` captured in an issue/story? I'm guessing yes, I just didn't see a link.

Reply:
I'm gonna do a full pass to ensure that they're all captured

Reply:
https://app.shortcut.com/revenuecat/story/11124/cache-sk2-products, will continue to add the others as I go along